frontend/src/components/Contexts/LanguageContext.tsx

```diff
@@ -0,0 +1,52 @@
+import { createContext, FC, useContext, useState } from 'react'
+import { dictionaryList } from '../../language'
+
+interface ILanguageContext {
+    language: string
+    textDictionary: { [text: string]: string }
+    switchLanguage: (newLanguage: string) => void
+}
+
+interface Props {
+    children: React.ReactNode
+}
+
+const defaultLanguage = {
+    language: 'en',
+    textDictionary: dictionaryList.en,
+    switchLanguage: (newLanguage: string) => {},
+}
+
+export const LanguageContext = createContext<ILanguageContext>(defaultLanguage)
+
+export const LanguageProvider: FC<Props> = ({ children }) => {
+    const prevLanguage = window.localStorage.getItem('rcml-lang')
+    const [language, setLanguage] = useState(prevLanguage || defaultLanguage.language)
+
+    const textDictionary =
+        language === 'en' || language === 'no' ? dictionaryList[language] : defaultLanguage.textDictionary
```

Comment:
Is there a better way to check if language is a valid key without hard-coding the keys?

Reply:
Maybe add config array for valid keys?

Reply:
If I try to do the check using a list, I get an error:
![image](https://user-images.githubusercontent.com/33518988/221566314-3782e3fd-ca81-4b68-8b57-f2e4d17a551b.png)

or
![image](https://user-images.githubusercontent.com/33518988/221567226-c267cf18-5ffa-4870-b8f3-1c07441aeaae.png)